tls/s2n_resume.c

```diff
@@ -1027,16 +1026,26 @@ int s2n_config_wipe_expired_ticket_crypto_keys(struct s2n_config *config, int8_t
 
 end:
     for (int j = 0; j < num_of_expired_keys; j++) {
-        POSIX_GUARD_RESULT(s2n_set_remove(config->ticket_keys, expired_keys_index[j] - j));
+        POSIX_GUARD_RESULT(s2n_array_remove(config->ticket_keys, expired_keys_index[j] - j));
     }
 
     return 0;
 }
 
 int s2n_config_store_ticket_key(struct s2n_config *config, struct s2n_ticket_key *key)
 {
-    /* Keys are stored from oldest to newest */
-    POSIX_GUARD_RESULT(s2n_set_add(config->ticket_keys, key));
+    uint32_t ticket_keys_len = 0;
+    POSIX_GUARD_RESULT(s2n_array_num_elements(config->ticket_keys, &ticket_keys_len));
+
+    /* The ticket key secret must be unique. */
+    for (uint32_t i = 0; i < ticket_keys_len; i++) {
```

Comment:
Any concerns around performance? s2n_set_add did a binary search, but we're now doing a linear search
https://github.com/aws/s2n-tls/blob/a694beeb81268bc066eb5edc2270b3cd34ff81f7/utils/s2n_set.c#L95-L96

Reply:
Oh I just noticed that part of the description:
> Performance wise, this is acceptable because S2N_MAX_TICKET_KEYS is 48.
Carry on :)